src/Umbraco.Web/Models/Mapping/ContentMapDefinition.cs

```diff
@@ -86,7 +86,10 @@ private static void Map(IContent source, ContentPropertyCollectionDto target, Ma
         // Umbraco.Code.MapAll -AllowPreview -Errors -PersistedContent
         private void Map(IContent source, ContentItemDisplay target, MapperContext context)
         {
-            target.AllowedActions = GetActions(source);
+            // Both GetActions and DetermineIsChildOfListView use parent, so get it once here
+            var parent = _contentService.GetParent(source);
```

Comment:
In theory this will all ready be looked up so we can avoid re-looking it back up again if it's in the context. In the content service the parent is explicitly looked up.

Reply:
Good idea, I've added the parent to the context, and look there before trying to use the service.